src/Data/Array/Mutable.idr

```diff
@@ -300,6 +300,33 @@ parameters {n : Nat}
         let x' # t := get r x t
           in x' # t
 
+--------------------------------------------------------------------------------
+--          Reversing Arrays
+--------------------------------------------------------------------------------
+
+parameters {n : Nat}
+           (p : MArray s n a)
+
+  ||| Reverse the order of elements in a mutable array.
+  export
+  mreverse : F1 s (MArray s n a)
+  mreverse t =
+    let trt # t := unsafeMArray1 n t
+     in go 0 n trt t
+    where
+      go :  (m, x : Nat)
+         -> (q : MArray s n a)
+         -> {auto v : Ix x n}
+         -> {auto 0 _ : LTE m $ ixToNat v}
+         -> {auto 0 _ : LTE x n}
+         -> F1 s (MArray s n a)
```

Comment:
The `m` argument is never used here and can be dropped. Other than that, this looks good.